src/detect-dns-query.c

```diff
@@ -199,6 +198,23 @@ static int PrefilterMpmDnsQueryRegister(DetectEngineCtx *de_ctx,
             pectx, PrefilterMpmDnsQueryFree, mpm_reg->pname);
 }
 
+#ifdef HAVE_LUA
+static int DetectEngineInspectDnsRequest(DetectEngineCtx *de_ctx, DetectEngineThreadCtx *det_ctx,
+        const struct DetectEngineAppInspectionEngine_ *engine, const Signature *s, Flow *f,
+        uint8_t flags, void *alstate, void *txv, uint64_t tx_id)
+{
+    return DetectEngineInspectGenericList(
+            de_ctx, det_ctx, s, engine->smd, f, flags, alstate, txv, tx_id);
+}
+
+static int DetectEngineInspectDnsResponse(DetectEngineCtx *de_ctx, DetectEngineThreadCtx *det_ctx,
+        const struct DetectEngineAppInspectionEngine_ *engine, const Signature *s, Flow *f,
+        uint8_t flags, void *alstate, void *txv, uint64_t tx_id)
+{
+    return DetectEngineInspectGenericList(
+            de_ctx, det_ctx, s, engine->smd, f, flags, alstate, txv, tx_id);
+}
+#endif
```

Comment:
These are only used for Lua scripting then?

Reply:
Indeed.
See also below previous comment :

> register these generic engines from here for now